Only link jemalloc for the Telegram binary

Some combination of factors is making the different codegen tools hang when
jemalloc is linked for those, and they're ran under portage's sandbox. Since
this is only used during build-time, and jemalloc is merely necessary to
improve runtime memory use, it's unnecessary to use it for anything else.

--- a/Telegram/CMakeLists.txt
+++ b/Telegram/CMakeLists.txt
@@ -1280,6 +1280,11 @@
         endif()
     endif()
 else()
+    target_link_libraries(Telegram
+    PRIVATE
+        desktop-app::external_jemalloc
+    )
+
     if (NOT DESKTOP_APP_DISABLE_DBUS_INTEGRATION)
         target_link_libraries(Telegram
         PRIVATE
--- a/cmake/options_linux.cmake
+++ b/cmake/options_linux.cmake
@@ -45,11 +45,6 @@
     target_link_options(common_options INTERFACE $<IF:$<CONFIG:Debug>,,-g -flto -fuse-linker-plugin>)
 endif()
 
-target_link_libraries(common_options
-INTERFACE
-    desktop-app::external_jemalloc
-)
-
 if (DESKTOP_APP_USE_ALLOCATION_TRACER)
     target_link_options(common_options
     INTERFACE
